frontends/web/src/routes/device/bitbox02/passphrase.tsx

```diff
@@ -35,345 +35,351 @@ const INFO_STEPS_DISABLE = 0;
 
 const CONTENT_MIN_HEIGHT = '38em';
 
-interface MnemonicPassphraseButtonProps {
+type TProps = {
     deviceID: string;
 }
 
-interface State {
-    infoStep: number;
-    passphraseEnabled?: boolean;
-    status: 'info' | 'progress' | 'success';
-    understood: boolean;
-}
-
-type Props = MnemonicPassphraseButtonProps & TranslateProps;
+type TStatus = 'info' | 'progress' | 'success';
 
-class Passphrase extends Component<Props, State> {
-  public readonly state: State = {
-    infoStep: 0,
-    status: 'info',
-    understood: false,
-  };
+export const Passphrase = ({ deviceID }: TProps) => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const [infoStep, setInfoStep] = useState(0);
+  const [status, setStatus] = useState<TStatus>('info');
+  const [passphraseEnabled, setPassphraseEnabled] = useState<boolean>();
 
-  public componentDidMount() {
-    getDeviceInfo(this.props.deviceID)
-      .then(result => {
-        if (!result.success) {
-          console.error(result.message);
-          alertUser(this.props.t('genericError'));
-          return;
-        }
-        const { mnemonicPassphraseEnabled } = result.deviceInfo;
-        this.setState({
-          // before enabling/disabling we show 1 or more pages to inform about the feature
-          // each page has a continue button that jumps to the next or finally toggles passphrase
-          // infoStep counts down in decreasing order
-          infoStep: mnemonicPassphraseEnabled
-            ? INFO_STEPS_DISABLE
-            : INFO_STEPS_ENABLE,
-          passphraseEnabled: mnemonicPassphraseEnabled,
-        });
-      })
-      .catch(console.error);
-  }
+  // Sync. component state with device info.
+  useEffect(() => {
+    getDeviceInfo(deviceID).then(result => {
+      if (!result.success) {
+        console.error(result.message);
+        alertUser(t('genericError'));
+        return;
+      }
+      const { mnemonicPassphraseEnabled } = result.deviceInfo;
+      // before enabling/disabling we show 1 or more pages to inform about the feature
+      // each page has a continue button that jumps to the next or finally toggles passphrase
+      // infoStep counts down in decreasing order
+      setPassphraseEnabled(mnemonicPassphraseEnabled);
+      setInfoStep(mnemonicPassphraseEnabled ? INFO_STEPS_DISABLE : INFO_STEPS_ENABLE);
+    }).catch(console.error);
+  }, [deviceID, setPassphraseEnabled, setInfoStep, t]);
 
-  private togglePassphrase = async () => {
-    const { deviceID, t } = this.props;
-    const enable = !this.state.passphraseEnabled;
-    this.setState({ status: 'progress' });
+  // Toggle Passphrase on device.
```

Comment:
maybe I can remove that comment, it is pretty clear what the function does...

Reply:
ping @NicolaLS plesae  remove those comments and rename the functions where needed.

Reply:
sorry, I was not sure if there were more change requests incomming and wanted to do it in one push. I removed the comment, but I did not rename any functions as `stopInfo` would be the only one and we said we'll rename it in the refactor that will follow after this PR.

Reply:
removed some more comments that were not really necessary. PTAL @thisconnect